backend/store/etcd/store.go

```diff
@@ -294,10 +294,31 @@ func namespaceFound(namespace string) clientv3.Cmp {
 	return keyFound(getNamespacePath(namespace))
 }
 
-func computeContinueToken(ctx context.Context, r corev2.Resource) (token string) {
-	objMeta := r.GetObjectMeta()
+// ComputeContinueToken calculates a continue token based on the given resource
+func ComputeContinueToken(ctx context.Context, r corev2.Resource) (token string) {
 	queriedNamespace := store.NewNamespaceFromContext(ctx)
 
+	switch resource := r.(type) {
+	case *corev2.Event:
+		// TODO(ccressent): This can surely be simplified
+		if queriedNamespace == "" {
+			// Workaround for sensu-go#2465: keepalive events do not always have
+			// their namespace filled in, which would break the construction of
+			// continue token below. To accommodate for that, when
+			// constructing the continue token, whevener an event has a
+			// namespace of "" we construct the continue token using its
+			// entity's namespace instead.
+			eventNamespace := resource.Namespace
+			if eventNamespace == "" {
+				eventNamespace = resource.Entity.Namespace
+			}
+			return "/" + eventNamespace + "/" + resource.Entity.Name + "/" + resource.Check.Name + "\x00"
+		}
+		return resource.Entity.Name + "/" + resource.Check.Name + "\x00"
+	}
+
+	objMeta := r.GetObjectMeta()
+
 	if queriedNamespace == "" {
 		token = path.Join(objMeta.Namespace, objMeta.Name) + "\x00"
 	} else {
```

Comment:
I would put the default, "regular" case in the switch as part of `default`.

To me that makes it clearer that there are distinct cases, and if a code path in `case *corev2.Event` is missing a `return`, it won't lead to some nonsensical operation being executed.